backend/src/main/java/saci/application/KnowledgeController.java

```diff
@@ -90,4 +93,15 @@ public ResponseEntity<Void> deleteKnowledgeById(@PathVariable long knowledgeId)
             return ResponseEntity.notFound().build();
         }
     }
+
+    @PutMapping("/{knowledgeId}")
+    public ResponseEntity<Knowledge> editKnowledge(
+            @PathVariable Long knowledgeId, @Valid @RequestBody Knowledge updatedKnowledge) {
+        try {
+            knowledgeService.editKnowledge(knowledgeId, updatedKnowledge);
+            return ResponseEntity.status(HttpStatus.ACCEPTED).build();
+        } catch (AlreadyExistsException ex) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).build();
```

Comment:
Now you have to catch 2 exceptions, where one must be Not Found and the other like it is (Conflict) so that in the frontend you know which error message to give.
To make the distinction between "knowledge name already exists" and "knowledge not found".

Reply:
You can use 
ResponseEntity.ok().build() instead of ACCEPTED
And also use ResponseEntity.notFound() for one of the exceptions.
I am not sure if there is a ResponseEntity.conflict() for the current exception but give it a check.